### ISSUE TITLE
Replace URIs in the JSON-LD output with blank nodes

### ISSUE THREAD

**Issue by gaurav** [phyloref/curation-tool#92]
The JSON-LD export implemented in #73 uses base URIs when exporting the PHYX file for reasoning. Instead of generating these URIs, it would be better if we could create them as [blank nodes](https://en.wikipedia.org/wiki/Blank_node) instead, since they are intended to be specific to this one ontology while reasoning.

I tried to implement this when writing #73, but blank nodes were not being reasoned over correctly and so I went with using `example.org` base URIs as an expediency. This issue covers either replacing those with blank nodes or showing definitively that they cannot be used in this way.

**Comment by hlapp** [comment]
I don't think using blank nodes is the right strategy. Instead, use document-local IDs, not global HTTP URIs. This is what Phenoscape does, for example (we use UUIDs).

@balhoff am I right with this?

**Comment by gaurav** [comment]
Ah, I hadn't thought of that! That definitely makes more sense and seems to work correctly with our existing infrastructure. I'll modify this issue to switch us to using document-local IDs instead.

**Comment by balhoff** [comment]
We do use UUIDs in some places, but when NeXML is converted to OWL they become HTTP URIs under a Phenoscape domain. However they are not dereferencable. Maybe it would be better for us to use the URN standard for UUIDs when there is no intention of dereferencing them: https://tools.ietf.org/html/rfc4122

**Comment by hlapp** [comment]
> Maybe it would be better for us to use the URN standard for UUIDs when there is no intention of dereferencing them: https://tools.ietf.org/html/rfc4122

Yeah, I agree. And that would also take care of complying with the NMTOKEN datatype requirement, if I recall that correctly?

**Comment by balhoff** [comment]
> And that would also take care of complying with the NMTOKEN datatype requirement, if I recall that correctly?

@hlapp I'm not following—I may have missed something.

**Comment by hlapp** [comment]
@balhoff wasn't that the issue with NeXML created by OntoTrace (or Phenex?) that would contain UUIDs as IDs which sometimes start with digits, which is disallowed by the NMTOKEN type?

**Comment by balhoff** [comment]
Oh, I think it was [NCName](https://www.w3.org/TR/xml-id/#processing). I will need to think about whether it works to use an absolute IRI in those places. Seems right, but for this XML stuff I always need to re-study it every time something like this comes up.
